src/assemble_workflow/dists.py

```diff
@@ -10,9 +10,10 @@
 
 
 class Dists:
+
     DISTRIBUTIONS_MAP = {
-        "tar": DistTar,
-        "zip": DistZip,
+        "tar": [DistTar, ".tar.gz"],
```

Comment:
The array causes you to do `DISTRIBUTIONS_MAP['tar'][0]` which doesn't really explain what this is. Turn it into a dictionary of dictionaries:

```python
"tar": {
    'klass': DistTar,
    'extension': ".tar.gz"
}
```

Access via `DISTRIBUTIONS_MAP['tar']['klass']`.

You can go further and make this dictionary a class and do

```python
DISTRIBUTIONS_MAP['tar'].klass`
```

Reply:
Thanks for the details I will do some test on this.
Thanks.